packages/jellyfish-api-core/src/category/blockchain.ts

```diff
@@ -17,6 +20,36 @@ export class Blockchain {
   async getBlockchainInfo (): Promise<BlockchainInfo> {
     return await this.client.call('getblockchaininfo', [], 'number')
   }
+
+  /**
+   * Get a hash of block in best-block-chain at height provided.
+   * @param height
+   * @return Promise<string>
+   */
+  async getBlockHash (height: number): Promise<string> {
+    return await this.client.call('getblockhash', [height], 'number')
+  }
+
+  /**
+   * Get the height of the most-work fully-validated chain.
+   * @return Promise<number>
+   */
+  async getBlockCount (): Promise<number> {
+    return await this.client.call('getblockcount', [], 'number')
+  }
+
+  /**
+   * Get block data with particular header hash.
+   * @param blockHash
+   * @param verbosity optional, default is 1, 0 for hex encoded
+   * @return Promise<BlockVerbo>
+   */
+  async getBlock (blockHash: string, verbosity: number): Promise<BlockVerbo2>
+  async getBlock (blockHash: string, verbosity: number): Promise<BlockVerbo1>
+  async getBlock (blockHash: string, verbosity: number): Promise<BlockVerbo0>
+  async getBlock (blockHash: string, verbosity: number): Promise<BlockVerbo> {
+    return await this.client.call('getblock', [blockHash, verbosity], 'number')
```

Comment:
```ts
async getBlock (blockHash: string, verbosity: 2): Promise<BlockVerbo2>
async getBlock (blockHash: string, verbosity: 1): Promise<BlockVerbo1>
async getBlock (blockHash: string, verbosity: 0): Promise<string>
async getBlock (blockHash: string, verbosity: 0 | 1 | 2): Promise<string | BlockBase> {
}
```

You can do something like this so that the type will always be correct based on the verbosity.

Reply:
```ts
type BlockVerbo0 = 'string'
export type BlockVerbo = BlockVerbo2 | BlockVerbo1 | BlockVerbo0 
```

will not be required when it's done as above.